lib/portus/registry_notification.rb

```diff
@@ -7,29 +7,26 @@ class RegistryNotification
     # An array with the events that a handler has to support.
     HANDLED_EVENTS = %w[push delete].freeze
 
-    # Processes the notification data with the given handlers. The data is the
-    # parsed JSON body as given by the registry. A handler is a class that can
-    # call the `handle_#{event}_event` method. This method receives an `event`
-    # object, which is the event object as given by the registry.
-    def self.process!(data, *handlers)
+    # It filters the event from the registry so the background job can actually
+    # handle this request.
+    def self.process!(data)
       data["events"].each do |event|
-        Rails.logger.debug "Incoming event:\n#{JSON.pretty_generate(event)}"
+        Rails.logger.debug "Filtering event:\n#{JSON.pretty_generate(event)}"
 
+        # Skip irrelevant or already-handled events.
         next unless should_handle?(event)
-        action = event["action"]
 
-        # Only register pushes, since they are the conflicting actions since 2.5
-        if action == "push"
-          RegistryEvent.create!(
-            event_id:   event["id"],
-            repository: event["target"]["repository"],
-            tag:        event["target"]["tag"]
-          )
-        end
+        # Only register relevant events.
+        next unless HANDLED_EVENTS.include? event["action"]
```

Comment:
Isn't this already being handled by `should_handle?`?

Reply:
That's true. Before it was slightly different, allowing only push actions, since delete actions where handled a bit differently. Let me test this out.

Reply:
Another question: is it okay to create a `RegistryEvent` for the delete actions? Before it was only for push.